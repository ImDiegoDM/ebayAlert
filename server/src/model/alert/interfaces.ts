import * as validator from 'validator'
import { ObjError } from '../../objError';

export interface SearchPhrase{
  phrase:string;
  howOften:2|10|30
}

export interface Alert{
  id?:string,
  email:string,
  phrases:SearchPhrase[]
}

export const collection = 'search_phrases';
