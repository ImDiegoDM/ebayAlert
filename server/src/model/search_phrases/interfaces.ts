export interface ISearchPhrase {
  email: string;
  phrase: string;
  howOften: 2 | 10 | 30;
}

export const collection = 'search_phrases';
