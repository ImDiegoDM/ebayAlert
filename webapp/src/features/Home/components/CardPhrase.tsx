import { faTrash, faTrashAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import styled from 'styled-components';
import { ISearchPhrase } from '../../../interfaces';

const Container = styled.div`
  width: 300px;
  height: 150px;
  position: relative;
  box-shadow: 0 0 10px 3px #b9b4b4;
  border-radius: 10px;
  margin: 10px 15px;
  cursor: pointer;
`;

const Email = styled.span`
  position:absolute;
  bottom:10px;
  left:10px;
`;

const HowOften = styled.span`
  position: absolute;
  bottom:10px;
  right:10px;
`;

const Phrase = styled.div`
  position: absolute;
  top:10px;
  left:10px;
  font-size: 1.7em;
  width: calc( 100% - 20px);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space:nowrap
`;

export interface CardPhraseProps {
  phrase: ISearchPhrase;
  onDelete?: () => void;
  onClick?: () => void;
}

const Icon = styled.div`
  position: absolute;
  top:10px;
  right:10px;
  color:red;
  cursor: pointer;
  z-index: 10;
`;

export function CardPhrase(props: CardPhraseProps) {
  return <Container onClick={props.onClick}>
    <Icon onClick={(e) => {
      e.stopPropagation();
      if (props.onDelete) {
        props.onDelete();
      }
    }}>
      <FontAwesomeIcon icon={faTrashAlt}/>
    </Icon>
    <Email>{props.phrase.email}</Email>
    <Phrase title={props.phrase.phrase}>{props.phrase.phrase}</Phrase>
    <HowOften>{props.phrase.howOften}</HowOften>
  </Container>;
}
