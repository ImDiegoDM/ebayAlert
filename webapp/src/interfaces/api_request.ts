import { AxiosError } from 'axios';

export interface IApiRequest<T= any, E= any> {
  fetching: boolean;
  fetched: boolean;
  ok: boolean;
  data?: T;
  error?: AxiosError;
}
