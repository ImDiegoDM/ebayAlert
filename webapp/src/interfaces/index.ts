export * from './api_request';
export * from './search_phrase';
