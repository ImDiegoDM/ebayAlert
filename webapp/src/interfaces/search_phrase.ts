export interface ISearchPhrase {
  phrase: string;
  email: string;
  howOften: '2' | '10' | '30';
}
