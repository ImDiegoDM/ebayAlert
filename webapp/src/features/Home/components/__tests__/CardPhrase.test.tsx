import { render } from '@testing-library/react';
import React from 'react';
import { ISearchPhrase } from '../../../../interfaces';
import { CardPhrase } from '../CardPhrase';

const phrase = {
  _id: 'sdfd5fa1dfad',
  email: 'test@test.com',
  howOften: '2',
  phrase: 'drone',
};

describe('<CardPrase/>', () => {
  it('should match snapshot', () => {
    const { container } = render(<CardPhrase phrase={phrase as ISearchPhrase}/>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
