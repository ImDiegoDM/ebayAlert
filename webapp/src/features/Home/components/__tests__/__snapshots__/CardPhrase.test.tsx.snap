// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardPrase/> should match snapshot 1`] = `
.c0 {
  width: 300px;
  height: 150px;
  position: relative;
  box-shadow: 0 0 10px 3px #b9b4b4;
  border-radius: 10px;
  margin: 10px 15px;
  cursor: pointer;
}

.c2 {
  position: absolute;
  bottom: 10px;
  left: 10px;
}

.c4 {
  position: absolute;
  bottom: 10px;
  right: 10px;
}

.c3 {
  position: absolute;
  top: 10px;
  left: 10px;
  font-size: 1.7em;
  width: calc( 100% - 20px);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1 {
  position: absolute;
  top: 10px;
  right: 10px;
  color: red;
  cursor: pointer;
  z-index: 10;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-trash-alt fa-w-14 "
      data-icon="trash-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
        fill="currentColor"
      />
    </svg>
  </div>
  <span
    class="c2"
  >
    test@test.com
  </span>
  <div
    class="c3"
    title="drone"
  >
    drone
  </div>
  <span
    class="c4"
  >
    2
  </span>
</div>
`;
